tests/test_basic.py

```diff
@@ -110,6 +111,112 @@ def test_histplot_density():
     return fig
 
 
+@pytest.mark.mpl_image_compare(style="default")
+def test_histplot_flow():
+    np.random.seed(0)
+    h = hist.Hist(hist.axis.Regular(20, 5, 15, name="x"), hist.storage.Weight())
+    h.fill(np.random.normal(10, 3, 400))
+    fig, axs = plt.subplots(1, 3, sharey=True, figsize=(15, 5))
+    axs = axs.flatten()
+
+    hep.histplot(h, ax=axs[0], flow="hint")
+    hep.histplot(h, ax=axs[1], flow="show")
+    hep.histplot(h, ax=axs[2], flow="sum")
+
+    axs[0].set_title("Default(hint)", fontsize=18)
+    axs[1].set_title("Show", fontsize=18)
+    axs[2].set_title("Sum", fontsize=18)
+    return fig
+
+
+@pytest.mark.mpl_image_compare(style="default")
+def test_histplot_hist_flow():
+    np.random.seed(0)
+    entries = np.random.normal(10, 3, 400)
+    h = hist.Hist(hist.axis.Regular(20, 5, 15, name="x"), hist.storage.Weight())
+    h2 = hist.Hist(hist.axis.Regular(20, 5, 15, name="x"), hist.storage.Weight())
+    h3 = hist.Hist(hist.axis.Regular(20, 5, 15, name="x"), hist.storage.Weight())
+    h4 = hist.Hist(hist.axis.Regular(20, 5, 15, name="x"), hist.storage.Weight())
+
+    h.fill(entries)
+    h2.fill(entries[entries < 15])
+    h3.fill(entries[entries > 5])
+    h4.fill(entries[(entries > 5) & (entries < 15)])
+    import uproot
+
+    f = uproot.recreate("flow_th1.root")
+    f["h"] = h
+    f["h2"] = h2
+    f["h3"] = h3
+    f["h4"] = h4
+    fig, axs = plt.subplots(2, 2, sharey=True, figsize=(10, 10))
+    axs = axs.flatten()
+
+    hep.histplot(h, ax=axs[0], flow="show")
+    hep.histplot(h2, ax=axs[1], flow="show")
+    hep.histplot(h3, ax=axs[2], flow="show")
+    hep.histplot(h4, ax=axs[3], flow="show")
+
+    axs[0].set_title("Two-side overflow", fontsize=18)
+    axs[1].set_title("Left-side overflow", fontsize=18)
+    axs[2].set_title("Right-side overflow", fontsize=18)
+    axs[3].set_title("No overflow", fontsize=18)
+    fig.subplots_adjust(hspace=0.2, wspace=0.2)
+    axs[0].legend()
+    return fig
+
+
+@pytest.mark.mpl_image_compare(style="default")
+def test_histplot_uproot_flow():
+    np.random.seed(0)
+    entries = np.random.normal(10, 3, 400)
+    import uproot
+
+    f = uproot.open("flow_th1.root")
+    h = f["h"]
+    h2 = f["h2"]
+    h3 = f["h3"]
+    h4 = f["h4"]
+
+    fig, axs = plt.subplots(2, 2, sharey=True, figsize=(10, 10))
+    axs = axs.flatten()
+
+    hep.histplot(h, ax=axs[0], flow="show")
```

Comment:
This still uses the hist object no? I am not sure if just passing f["h"] will pass the TH1 or if you need to close and load the file, can you check?

Reply:
No, this use numpy hist

Reply:
passing f["h"] will pass the TH1